include/hit_record.hpp

```diff
@@ -0,0 +1,25 @@
+#ifndef HIT_RECORD_HPP
+#define HIT_RECORD_HPP
+
+#include "primitives.hpp"
+
+namespace raytracer::visitor {
+class hit_record {
+ public:
+  real_t t;
+  point p;         // hit point
+  vec normal;      // normal at hit point
+  bool front_face; // to check if hit point is on the outer surface
+  /*local coordinates for rectangles
+  and mercator coordinates for spheres */
+  real_t u;
+  real_t v;
+
+  // To set if the hit point is on the front face
```

Comment:
While you are here, add more Doxygen
```suggestion
  point p;         ///< Hit point
  vec normal;      ///< Normal at hit point
  bool front_face; ///< To check if hit point is on the outer surface
  /// Local coordinates for rectangles and mercator coordinates for spheres
  real_t u;
  real_t v;

  /// To set if the hit point is on the front face
```